utils/hardware_info.go

```diff
@@ -0,0 +1,61 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"go.viam.com/utils"
+)
+
+// GetArchitectureInfo returns the architecture of the board.
+func GetArchitectureInfo(modelName string) (utils.StringSet, error) {
+	arch := runtime.GOARCH
+
+	switch {
+	case strings.HasPrefix(arch, "amd"):
+		return stringSetFromX86(modelName)
+	case strings.HasPrefix(arch, "arm"):
+		return stringSetFromARM(modelName)
+	default:
+		return nil, noBoardError(modelName)
+	}
+}
+
+// A helper function for ARM architecture to process contents of a
+// given content of a file from os.ReadFile.
+func stringSetFromARM(modelName string) (utils.StringSet, error) {
+	path := "/proc/device-tree/compatible"
+	compatiblesRd, err := os.ReadFile(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, noBoardError(modelName)
+		}
+		return nil, err
+	}
+
+	return utils.NewStringSet(strings.Split(string(compatiblesRd), "\x00")...), nil
+}
+
+// A helper function for AMD architecture to process contents of a
+// given content of a file from os.ReadFile.
+func stringSetFromX86(modelName string) (utils.StringSet, error) {
+	path := "/sys/devices/virtual/dmi/id/board_name"
```

Comment:
Part of me says this should be `const`, and part of me says it's fine as-is. Same comment on line 30.

Reply:
yes! this should be constant.